### PR TITLE
feat: rollup everything

### LINKED ISSUE #862: Allow ES6 + use Babel for ES5 support
Vega seems to be using a lot of ES6 features, such as object destructuring assignments, and yet it use uglifyjs, not uglifyjs-es6. Also, there is a note that IE10 & 11 are also supported [here](https://vega.github.io/vega/usage/#ie), but IE11 does not support many other features either. According to http://gs.statcounter.com/, IE usage is at 4% and dropping. Is it even worth supporting and jumping through tons of hoops?

### PR DESCRIPTION
**Pull request by domoritz**
* Single rollup config for all modules
  * Consistency across modules
  * Extensible config file
* We can now use modern javascript since code goes through babel. We use a browserlist to ensure compatibility.
* no new major dependencies
* The browser bundles can be smaller iife bundles
* Sourcemaps for all bundles
* Fixed potential bugs:
  * vega-projection-extended was externalizing d3 array in core but we use a different version of d3 array that is not part of d3 (https://github.com/vega/vega/blob/a8e6991411c2c5c4805320f87276e9861e80d69c/packages/vega/rollup.js#L13).

Fixes #862

See #1165

### CODE REVIEW

**Comment by jheer** [comment]
This looks pretty nice! Thanks!! However, we will need to update the PR in response to changes from other PRs. This includes a change to the excluded D3 modules for `vega-core`, thanks to the D3 v6 release.

Also, I noticed vega.min.js replacing vega.js in a bunch of places. The use of the unminified bundle was intentional to aid debugging. Is the thought here that we should use the minified code + source map instead?

**Comment by domoritz** [comment]
Awesome. I'll take a look at upgrading the pull request. 

Yes, I thought we could use the Minifie's file with source maps. With this change, we would be testing that the source maps work. I'm happy to go back to the unminified version.

File: packages/vega-canvas/package.json [hunk 1]
```diff
@@ -10,18 +10,17 @@
...
-    "prebuild": "rimraf build && mkdir build",
-    "build": "yarn rollup",
-    "pretest": "yarn prebuild && yarn rollup",
+    "prebuild": "rimraf build",
+    "build": "rollup -c --config-browser",
+    "pretest": "yarn build --config-test",
     "test": "tape 'test/**/*-test.js'",
     "prepublishOnly": "yarn test && yarn build",
-    "postpublish": "git push && git push --tags"
+    "postpublish": "git push"
```
Comment (domoritz) [inline packages/vega-canvas/package.json:23]: Are you still using the postpublish command @jheer?

Comment (domoritz) [reply packages/vega-canvas/package.json]: I'm removing it.

**Comment by domoritz** [comment]
Done updating.

**Comment by domoritz** [comment]
I like this kind of setup that centralizes the build logic in Rollup and started doing the same for other projects like Vega-Embed: https://github.com/vega/vega-embed/pull/544.

**Pull request merged**